src/WebJobs.Script.WebHost/FunctionRequestInvoker.cs

```diff
@@ -0,0 +1,122 @@
+using Microsoft.Azure.WebJobs.Script.WebHost.WebHooks;
+
+namespace Microsoft.Azure.WebJobs.Script.Host
+{
+    public static class FunctionRequestInvoker
+    {
+        private static bool IsHomepageDisabled
+        {
+            get
+            {
+                return string.Equals(Environment.GetEnvironmentVariable(EnvironmentSettingNames.AzureWebJobsDisableHomepage),
+                    bool.TrueString, StringComparison.OrdinalIgnoreCase);
+            }
+        }
+
+        public static async Task<HttpResponseMessage> ProcessRequestAsync(HttpRequestMessage request, FunctionDescriptor function, CancellationToken cancellationToken, WebScriptHostManager scriptHostManager, WebHookReceiverManager webHookReceiverManager)
+        {
+            var httpTrigger = function.GetTriggerAttributeOrNull<HttpTriggerAttribute>();
+            bool isWebHook = !string.IsNullOrEmpty(httpTrigger.WebHookType);
+            var authorizationLevel = request.GetAuthorizationLevel();
+            HttpResponseMessage response = null;
+
+            if (isWebHook)
+            {
+                if (request.HasAuthorizationLevel(AuthorizationLevel.Admin))
+                {
+                    // Admin level requests bypass the WebHook auth pipeline
+                    response = await scriptHostManager.HandleRequestAsync(function, request, cancellationToken);
+                }
+                else
+                {
+                    // This is a WebHook request so define a delegate for the user function.
+                    // The WebHook Receiver pipeline will first validate the request fully
+                    // then invoke this callback.
+                    Func<HttpRequestMessage, Task<HttpResponseMessage>> invokeFunction = async (req) =>
+                    {
+                        // Reset the content stream before passing the request down to the function
+                        Stream stream = await req.Content.ReadAsStreamAsync();
+                        stream.Seek(0, SeekOrigin.Begin);
+
+                        return await scriptHostManager.HandleRequestAsync(function, req, cancellationToken);
+                    };
+                    response = await webHookReceiverManager.HandleRequestAsync(function, request, invokeFunction);
+                }
+            }
+            else
+            {
+                // Authorize
+                if (!request.HasAuthorizationLevel(httpTrigger.AuthLevel))
+                {
+                    return new HttpResponseMessage(HttpStatusCode.Unauthorized);
+                }
+
+                // Not a WebHook request so dispatch directly
+                response = await scriptHostManager.HandleRequestAsync(function, request, cancellationToken);
+            }
+
+            return response;
+        }
+
+        public static async Task<HttpResponseMessage> PreprocessRequestAsync(HttpRequestMessage request, FunctionDescriptor function, IDependencyResolver resolver)
```

Comment:
>IDependencyResolver [](start = 134, length = 19)

Passing the depednency resolver around seems suspect ... can we  pass just the interfaces we need (ISecretManager?) .
This is a static method; could this be an instance method and then the require interfaces are properties?